### PR TITLE
Code style options page UI updates

### PR DESCRIPTION
**Pull request by balajikris**
Fixes the following minor glitches:
1. preview does not update if selected row in the grid changes.
2. No preview if no row is selected
3. generic preview text and typos.

There aren't any known work items tracking these bugs.

### CODE REVIEW

**Comment by balajikris** [comment]
@dotnet/roslyn-ide

File: src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml.cs [hunk 1]
```diff
@@ -29,7 +30,7 @@ private void Options_SelectionChanged(object sender, SelectionChangedEventArgs e
             var dataGrid = (DataGrid)sender;
             var codeStyleItem = (AbstractCodeStyleOptionViewModel)dataGrid.SelectedItem;
 
-            if (codeStyleItem != null && codeStyleItem.NotificationsAvailable)
+            if (codeStyleItem != null)
```
Comment (brettfo) [inline src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml.cs]: ?.

**Comment by brettfo** [comment]
:shipit:, particularly for adding better samples for `this.`/`Me.`.

File: src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml.cs (hunk 1 above)
Comment (balajikris) [inline src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml.cs]: yep, that bad. it shouldn't have been checked in in the first place. I was testing something locally and unintentionally shipped it in the preview, the resulting bug is that if severity levels weren't available, previews will not be updated if user clicks on a different code style in the grid.

**Comment by jmarolf** [comment]
:+1:

**Comment by balajikris** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/5871/testReport/junit/Microsoft.CodeAnalysis.UnitTests.Interactive/InteractiveHostTests/StackOverflow/
// Retest reason: Test failure is a dupe of https://github.com/dotnet/roslyn/issues/10323 which is now fixed in master and this run is against future branch. So, no further action is necessary.

**Comment by balajikris** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/5877/
// Retest reason: Test failure is a dupe of #10323

**Comment by balajikris** [comment]
the interactive test keeps failing, I got a verbal okay from @tmat to merge with it.

File: src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs [hunk 2]
```diff
@@ -29,24 +27,92 @@ internal override bool ShouldPersistOption(OptionKey key)
 
         #region "Preview Text"
 
-        private static readonly string s_declarationPreviewTrue = @"
+        private static readonly string s_fieldDeclarationPreviewTrue = @"
+class C{
+    int capacity;
+    void foo()
```
Comment (Pilchie) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs]: Let's avoid using `foo` in option descriptions.  Also, I'd follow style guidelines for these.  How about `Method()`?

Comment (Pilchie) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs]: Ah, I see it's not visible, but still...

File: src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs [hunk 3]
```diff
@@ -29,24 +27,92 @@ internal override bool ShouldPersistOption(OptionKey key)
...
-        x = 0;
+        Id = 0;
+//]
+    }
+}";
+
+        private static readonly string s_eventDeclarationPreviewTrue = @"
+using System;
+class C{
+    event EventHandler e;
```
Comment (Pilchie) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs]: Most events are named with upper-case names, I'd follow that here.

**Comment by Pilchie** [comment]
:+1: modulo naming nits.

**Pull request merged**